File: spring-boot-test-slice-annotations/src/main/java/de/rieckpil/blog/Book.java
package de.rieckpil.blog;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import org.hibernate.annotations.NaturalId;

@Entity
public class Book {

  @Id @GeneratedValue private Long id;

  @NaturalId private String isbn;

  @Column(nullable = false)
  private String title;

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getIsbn() {
    return isbn;
  }

  public void setIsbn(String isbn) {
    this.isbn = isbn;
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }
}

File: spring-boot-kotlin-testcontainers/src/main/kotlin/de/rieckpil/blog/Book.kt
package de.rieckpil.blog

import org.hibernate.annotations.NaturalId
import javax.persistence.*

@Entity
@Table(name = "books")
class Book(
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  var id: Long,

  @NaturalId
  var isbn: String,

  @Column(nullable = false)
  var title: String
)
